crates/solvers/src/domain/dex/slippage.rs

```diff
@@ -95,72 +98,67 @@ impl Slippage {
         // internally casting its `BigInt` digits to a `i128` and unwrapping.
         // This means that there is a maximum of 38-digits of precision
         // (specifically, `i128::MAX.to_string().len() - 1`) allowed when
-        // rounding. So, in order to be pragmatic (and seing that 38 digits of
+        // rounding. So, in order to be pragmatic (and seeing that 38 digits of
         // precision is more than enough for slippage), first truncate the
-        // value to the maximum preicision and then round.
+        // value to the maximum precision and then round.
         Self(self.0.with_prec(38).round(arg as _))
     }
 }
 
-/// Token reference prices for a specified auction.
-pub struct Prices(pub HashMap<eth::TokenAddress, BigDecimal>);
-
-impl Prices {
-    pub fn new(prices: impl IntoIterator<Item = (eth::TokenAddress, auction::Price)>) -> Self {
-        Self(
-            prices
-                .into_iter()
-                .map(|(token, price)| (token, conv::ether_to_decimal(&price.0)))
-                .collect(),
-        )
-    }
-
-    /// Computes the set of reference prices for the specified auction.
-    pub fn for_auction(auction: &auction::Auction) -> Self {
-        Self::new(
-            auction
-                .tokens
-                .iter()
-                .filter_map(|(address, token)| Some((*address, token.reference_price?))),
-        )
-    }
-
-    /// Returns the reference price, in Ether for the specified token.
-    pub fn reference_price(&self, token: eth::TokenAddress) -> Option<auction::Price> {
-        self.0
-            .get(&token)
-            .and_then(conv::decimal_to_ether)
-            .map(auction::Price)
-    }
-}
-
 #[cfg(test)]
 mod tests {
-    use super::*;
+    use {
+        super::*,
+        crate::domain::auction::{self, Auction},
+    };
 
+    // TODO Notice that I'm forced to change the tests only because I refactored
```

Comment:
If you change an interface of a module in `driver`, and refactor the code without changing behavior, you will not need to change any tests.

Reply:
In general, I also think it is annoying to have to modify unit tests for refactors (it causes unnecessary additional barriers to refactoring that may dissuade people from doing the refactor they want for the additional work).

On the other hand, in this particular case I think its fine - it just caused changes to unit tests in the same module that was refactored (so, it can

In theory this _can_ be tested at the `solvers` boundary, it would be a lot harder though (and arguably less "correct", as integration tests assertions may require assumptions about the details of the implementation).